test/orange/orangeinp/detail/LogicUtils.test.cc

```diff
@@ -0,0 +1,97 @@
+//------------------------------- -*- C++ -*- -------------------------------//
+// Copyright Celeritas contributors: see top-level COPYRIGHT file for details
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file orange/orangeinp/detail/LogicUtils.test.cc
+//---------------------------------------------------------------------------//
+#include "orange/orangeinp/detail/LogicUtils.hh"
+
+#include <string_view>
+#include <vector>
+
+#include "corecel/cont/Span.hh"
+#include "orange/OrangeTypes.hh"
+#include "orange/detail/OrangeInputIOImpl.json.hh"
```

Comment:
Instead of this detail include and the JSON requirement, what about moving `string_to_logic` and `logic_to_string` to `orangeinp/LogicUtils.{hh,cc}` as well as putting the infix conversion there?